src/compiler/checker.ts

```diff
@@ -2719,7 +2728,8 @@ module ts {
                     var type = getUnionType(map(signature.unionSignatures, getReturnTypeOfSignature));
                 }
                 else {
-                    var type = getReturnTypeFromBody(<FunctionLikeDeclaration>signature.declaration);
+                    var declaration = <FunctionLikeDeclaration>signature.declaration;
+                    var type = getReturnTypeFromBody(declaration);

```

Comment:
Is this just to clarify the flow of things? I'm fine with it but I'm wondering if it serves a different purpose.